src/common/fundFromTurbo.ts

```diff
@@ -0,0 +1,252 @@
+  purchaseReceipt: ArNSPurchaseReceipt;
+};
+
+export interface FundFromTurboInterface {
+  getArNSPurchasePrice(params: InitiateArNSPurchaseParams): Promise<number>;
+  initiateArNSPurchase(
+    params: InitiateArNSPurchaseParams,
+    options: WriteOptions,
+  ): Promise<ArNSPurchaseResult>;
+  getArNSPurchaseReceipt(intent: string): Promise<ArNSPurchaseReceipt>;
+}
+
+export class FundFromTurbo {
+  private readonly paymentUrl: string;
+  private readonly axios: AxiosInstance;
+  private readonly logger: ILogger;
+  private readonly signer?: FundFromTurboSigner;
+
+  constructor({
+    paymentUrl = 'https://payment.ardrive.io',
+    axios = createAxiosInstance(),
+    logger = Logger.default,
+    signer,
+  }: TurboConfig) {
+    this.paymentUrl = paymentUrl;
+    this.axios = axios;
+    this.logger = logger;
+    this.signer = signer;
+  }
+
+  public async getArNSPurchasePrice({
+    intent,
+    name,
+    increaseQty,
+    type,
+    years,
+  }: InitiateArNSPurchaseParams): Promise<{ winc: string; mARIO: mARIOToken }> {
+    const url = new URL(`${this.paymentUrl}/v1/arns/price/${intent}/${name}`);
+    if (increaseQty !== undefined) {
+      url.searchParams.append('increaseQty', increaseQty.toString());
+    }
+    if (type !== undefined) {
+      url.searchParams.append('type', type);
+    }
+    if (years !== undefined) {
+      url.searchParams.append('years', years.toString());
+    }
+    const { data, status } = await this.axios.get<{
+      winc: string;
+      mARIO: string;
+    }>(url.toString());
+
+    this.logger.debug('getArNSPurchasePrice', {
+      intent,
+      name,
+      increaseQty,
+      type,
+      years,
+      data,
+      status,
+    });
+
+    if (status !== 200) {
+      throw new Error(
+        'Failed to get ArNS purchase price' + JSON.stringify(data),
+      );
+    }
+    if (!data.winc || !data.mARIO) {
+      throw new Error('Invalid response from Turbo' + JSON.stringify(data));
+    }
+
+    return {
+      winc: data.winc,
+      mARIO: new mARIOToken(+data.mARIO),
+    };
+  }
+
+  public async initiateArNSPurchase(
+    {
+      intent,
+      name,
+      increaseQty,
+      processId,
+      type,
+      years,
+    }: InitiateArNSPurchaseParams,
+    options?: WriteOptions,
+  ): Promise<ArNSPurchaseResult> {
+    if (!this.signer) {
+      throw new Error(
+        'Signer required for initiating ArNS purchase with Turbo',
+      );
+    }
+
+    const tags = [
+      { name: 'Turbo-ArNS-Purchase-Intent', value: intent },
+      { name: 'Name', value: name },
+      { name: 'Quantity', value: increaseQty?.toString() },
+      { name: 'Type', value: type },
+      { name: 'Years', value: years?.toString() },
+      { name: 'Process-Id', value: processId },
+    ];
+    const prunedTags = pruneTags(tags);
+    const nonce = v4();
+
+    if (options && options.tags) {
+      options.tags.forEach((tag) => {
+        prunedTags.push(tag);
+      });
+    }
```

Comment:
recommend pushing the tags then pruning - will handle invalid tags in the options.

```suggestion
    const tags = [
      { name: 'Turbo-ArNS-Purchase-Intent', value: intent },
      { name: 'Name', value: name },
      { name: 'Quantity', value: increaseQty?.toString() },
      { name: 'Type', value: type },
      { name: 'Years', value: years?.toString() },
      { name: 'Process-Id', value: processId },
      ...(options?.tags || [])
    ];
    const prunedTags = pruneTags(tags);
    const nonce = v4();
```

Reply:
removed this, we weren't using the tags anymore in signed req approach